### PR TITLE
Add QXmlEdit

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions.
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/lbellonda/qxmledit/issues/77
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. (https://github.com/lbellonda/qxmledit/pull/84 https://github.com/lbellonda/qxmledit/pull/85)

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.qxmledit.qxmledit.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      - install -Dm0644 src/images/icon.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
+      - mv ${FLATPAK_DEST}/share/applications/QXmlEdit.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - desktop-file-edit --set-key="Icon" --set-value="${FLATPAK_ID}" "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+      - mv ${FLATPAK_DEST}/share/appdata/QXmlEdit.appdata.xml ${FLATPAK_DEST}/share/appdata/${FLATPAK_ID}.appdata.xml
+    sources:
+      - type: git
+        url: https://github.com/lbellonda/qxmledit.git
+        tag: 0.9.16
+        commit: 5a06fbcc648500fc0e4c813ffe5d774fa332e6bc
+        dest: qxmledit
```
Comment (hfiguiere) [inline org.qxmledit.qxmledit.yml:40]: you shouldn't need to do that (and line 16)

File: org.qxmledit.qxmledit.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      - QXMLEDIT_INST_DESKTOP_DIR=${FLATPAK_DEST}/share/applications/
+      - QXMLEDIT_INST_APPDATA_DIR=${FLATPAK_DEST}/share/appdata/
+      - QXMLEDIT_INST_DIR=${FLATPAK_DEST}/bin/
+      - QXMLEDIT_INST_DOC_DIR=${FLATPAK_DEST}/share/doc/qxmledit
+      - QXMLEDIT_INST_LIB_DIR=${FLATPAK_DEST}/lib/
+      - QXMLEDIT_INST_TRANSLATIONS_DIR=${FLATPAK_DEST}/share/qxmledit/translations/
+    post-install:
+      - install -Dm0644 install_scripts/environment/icon/qxmledit_48x48.png ${FLATPAK_DEST}/share/icons/hicolor/48x48/apps/${FLATPAK_ID}.png
+      - install -Dm0644 src/images/icon.svg ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/${FLATPAK_ID}.svg
+      - mv ${FLATPAK_DEST}/share/applications/QXmlEdit.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
```
Comment (hfiguiere) [inline org.qxmledit.qxmledit.yml:32]: you should just use `rename-desktop-file: QXmlEdit.desktop`

File: org.qxmledit.qxmledit.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      - desktop-file-edit --set-key="Icon" --set-value="${FLATPAK_ID}" "${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop"
+      - mv ${FLATPAK_DEST}/share/appdata/QXmlEdit.appdata.xml ${FLATPAK_DEST}/share/appdata/${FLATPAK_ID}.appdata.xml
```
Comment (hfiguiere) [inline org.qxmledit.qxmledit.yml:34]: yui should just ise `rename-appdata-file: QXmlEdit.appdata.xml`

**Comment by hfiguiere** [comment]
bot, build org.qxmledit.QXmlEdit

**Comment by hfiguiere** [comment]
The manifest file should be named after the app-id: `org.qxmledit.QXmlEdit.json`. Otherwise it won't be found by the flathub bot.

**Comment by Mailaender** [comment]
Updated. Can I keep it in @yaml?

**Comment by JakobDev** [comment]
The Appdata patches should not been needed. Everything is valid according to the [Specification](https://www.freedesktop.org/software/appstream/docs/). The thing, that should be changed, is that the Mimetypes tag is [depracted](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-mimetypes). But it is still valid and flatpak-builder should not complain about it. Maybe you have a old version of appstream-glib.

**Comment by Mailaender** [comment]
Removed the appdata patch. I got this gibberish error message on https://neon.kde.org/ without them:
```
Error loading AppData file: failed to parse /app/share/appdata/org.qxmledit.QXmlEdit.appdata.xml: Error on line 25 char 9: <categories> already set '' and tried to replace with '
    '
```

**Comment by hfiguiere** [comment]
> Updated. Can I keep it in @yaml?

sorry it's about it being case sensitive. I mistakenly wrote .json.

**Comment by hfiguiere** [comment]
bot, build org.qxmledit.QXmlEdit

File: org.qxmledit.QXmlEdit.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        url: https://github.com/lbellonda/qxmledit.git
+        tag: 0.9.16
+        commit: 5a06fbcc648500fc0e4c813ffe5d774fa332e6bc
+        x-checker-data:
+          type: git
+          tag-pattern: ^([\d.]+)$
+      - type: patch
+        path: install.patch
+      - type: patch
+        path: categories.patch
```
Comment (hfiguiere) [inline org.qxmledit.QXmlEdit.yml:46]: this patch was removed from the repo, so it is not found.

(hence the failure)

**Comment by hfiguiere** [comment]
bot, build org.qxmledit.QXmlEdit

**Comment by Mailaender** [comment]
> Error loading AppData file: failed to parse /srv/buildbot/worker/build-x86_64-6/build/.flatpak-builder/rofiles/rofiles-GLbTvX/files/share/appdata/org.qxmledit.QXmlEdit.appdata.xml: Error on line 24 char 9: <categories> already set '​' and tried to replace with '
    ​'
Error: ERROR: appstream-compose failed: Child process exited with code 1

Same problem I encountered, so I am adding the patch back in.

**Comment by hfiguiere** [comment]
bot, build org.qxmledit.QXmlEdit

**Comment by hfiguiere** [comment]
```
builddir/files/share/appdata/org.qxmledit.QXmlEdit.appdata.xml: FAILED:
• tag-missing           : <content_rating> required [use https://odrs.gnome.org/oars]
• attribute-invalid     : <screenshot> width (800) did not match specified (808) [https://raw.githubusercontent.com/lbellonda/qxmledit/master/install_scripts/environment/desktop/screenshots/qxmledit800.png]
• attribute-invalid     : <screenshot> height (600) did not match specified (599) [https://raw.githubusercontent.com/lbellonda/qxmledit/master/install_scripts/environment/desktop/screenshots/qxmledit800.png]
• attribute-missing     : <release> has no timestamp
Validation of files failed
```

**Comment by Mailaender** [comment]
Updated my patch https://github.com/lbellonda/qxmledit/pull/85.

**Comment by hfiguiere** [comment]
btw you can also trigger the build


bot, build org.qxmledit.QXmlEdit

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Pull request closed without merging**